3.10.0 release build paralyzation issue

👋 looks like there is some build parallelization issue with 3.10.0 release build

```
  echo "unset LUA_PATH LUA_PATH_5_2 LUA_PATH_5_3 LUA_PATH_5_4 LUA_CPATH LUA_CPATH_5_2 LUA_CPATH_5_3 LUA_CPATH_5_4" >> luarocks-admin
  install: ./build/config-5.4.lua: No such file or directory
  make: *** [install-config] Error 71
```

see build log in here, https://github.com/Homebrew/homebrew-core/actions/runs/8208660378/job/22452665344
relates to https://github.com/Homebrew/homebrew-core/pull/165391

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -1,3 +1,4 @@
+MAKEFLAGS += --jobs=1
 
 -include config.unix
 

